android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateUtils.java

```diff
@@ -1,13 +1,70 @@
 package com.microsoft.codepush.react;
 
+import android.util.Base64;
+
 import com.facebook.react.bridge.ReadableArray;
 import com.facebook.react.bridge.WritableMap;
 
+import org.json.JSONArray;
+
+import java.io.ByteArrayInputStream;
 import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
 import java.io.IOException;
+import java.io.InputStream;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Collections;
 
 public class CodePushUpdateUtils {
 
+    private static void addContentsOfFolderToManifest(String folderPath, String pathPrefix, ArrayList<String> manifest) {
+        File folder = new File(folderPath);
+        File[] folderFiles = folder.listFiles();
+        for (File file : folderFiles) {
+            String fullFilePath = file.getAbsolutePath();
+            String relativePath = (pathPrefix.isEmpty() ? "" : (pathPrefix + "/")) + file.getName();
+            if (file.isDirectory()) {
+                addContentsOfFolderToManifest(fullFilePath, relativePath, manifest);
+            } else {
+                try {
+                    // Substring 1 because appendPathComponent creates a "/" prefix to the relativePath.
+                    manifest.add(relativePath + ":" + computeHash(new FileInputStream(file)));
```

Comment:
oops! that was leftover from when I was trying to iterate on the code. will remove.